File: SourceCode/PPTControllerClient-Android/src/com/pptcontroller/util/InputValidator.java
package com.pptcontroller.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class InputValidator {
	private static String regexIPAddress = "\\b((?!\\d\\d\\d)\\d+|1\\d\\d|2[0-4]\\d|25[0-5])\\.((?!\\d\\d\\d)\\d+|1\\d\\d|2[0-4]\\d|25[0-5])\\.((?!\\d\\d\\d)\\d+|1\\d\\d|2[0-4]\\d|25[0-5])\\.((?!\\d\\d\\d)\\d+|1\\d\\d|2[0-4]\\d|25[0-5])\\b";
    private static String regexPort = "^\\d+$";

    public static boolean ValidateIpAddress(String ipAddress)
    {
    	Pattern pattern = Pattern.compile(regexIPAddress);
    	Matcher match = pattern.matcher(ipAddress);
        return match.matches();
    }

    public static boolean ValidatePort(String port)
    {
    	Pattern pattern = Pattern.compile(regexPort);
    	Matcher match = pattern.matcher(port);
        return match.matches();
    }
}

File: SourceCode/PPTControllerClient-WindowsPhone/PPTControler/Tasks/InputValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace PPTController.Tasks
{
    public class InputValidator
    {
        private Regex regexIPAddress = new Regex(@"^\d+\.\d+\.\d+\.\d+$");
        private Regex regexPort = new Regex(@"^\d+$");

        public bool ValidateIpAddress(string ipAddress)
        {
            return regexIPAddress.IsMatch(ipAddress);
        }

        public bool ValidatePort(string port)
        {
            return regexPort.IsMatch(port);
        }
    }
}
